pkg/component/logging/vali/vali.go

```diff
@@ -0,0 +1,1104 @@
+	fsGroupChangeOnRootMismatch = corev1.FSGroupChangeOnRootMismatch
+	pathType                    = networkingv1.PathTypePrefix
+
+	ingressTLSCertificateValidity = 730 * 24 * time.Hour // ~2 years, see https://support.apple.com/en-us/HT210176
+)
+
+func init() {
+	var err error
+	valiConfigTmpl, err = template.
+		New(tplNameValiConfig).
+		Funcs(sprig.TxtFuncMap()).
+		Parse(tplContentValiConfig)
+	if err != nil {
+		panic(err)
+	}
+
+	telegrafConfigTmpl, err = template.
+		New(tplNameTelegrafStart).
+		Funcs(sprig.TxtFuncMap()).
+		Parse(tplContentTelegrafConfig)
+	if err != nil {
+		panic(err)
+	}
+
+	telegrafStartTmpl, err = template.
+		New(tplNameTelegrafConfig).
+		Funcs(sprig.TxtFuncMap()).
+		Parse(tplContentTelegrafStart)
+	if err != nil {
+		panic(err)
+	}
+}
+
+// Values are the values for the Vali.
+type Values struct {
+	Replicas              int32
+	AuthEnabled           bool
+	Storage               *resource.Quantity
+	HvpaEnabled           bool
+	MaintenanceBegin      string
+	MaintenanceEnd        string
+	ValiImage             string
+	CuratorImage          string
+	RenameLokiToValiImage string
+	InitLargeDirImage     string
+	PriorityClassName     string
+	TelegrafImage         string
+	KubeRBACProxyImage    string
+	ClusterType           string
+	IngressClass          string
+	ValiHost              string
+}
+
+type vali struct {
+	client         client.Client
+	namespace      string
+	secretsManager secretsmanager.Interface
+	k8sVersion     *semver.Version
+	values         Values
+}
+
+var _ component.Deployer = &vali{}
+
+// New creates a new instance of Vali deployer.
+func New(
+	client client.Client,
+	namespace string,
+	secretsManager secretsmanager.Interface,
+	k8sVersion *semver.Version,
+	values Values) (component.Deployer, error) {
+
+	switch {
+	case client == nil:
+		return nil, errors.New("client cannot be nil")
+	case len(namespace) == 0:
+		return nil, errors.New("namespace cannot be empty")
+	case len(values.ValiImage) == 0:
+		return nil, errors.New("vali image cannot be empty")
+	case len(values.CuratorImage) == 0:
+		return nil, errors.New("vali curator image cannot be empty")
+	case len(values.InitLargeDirImage) == 0:
+		return nil, errors.New("init-large-dir image cannot be empty")
+	case len(values.KubeRBACProxyImage) > 0 && len(values.TelegrafImage) == 0:
+		return nil, errors.New("when kube-RBAC-proxy image is present the telegraf image cannot be empty")
+	case len(values.TelegrafImage) > 0 && len(values.KubeRBACProxyImage) == 0:
+		return nil, errors.New("when telegraf image is present the kube-RBAC-proxy image cannot be empty")
+	case len(values.KubeRBACProxyImage) > 0 && len(values.TelegrafImage) > 0 && secretsManager == nil:
+		return nil, errors.New("secretManager cannot be nil")
+	case len(values.KubeRBACProxyImage) > 0 && len(values.TelegrafImage) > 0 && k8sVersion == nil:
+		return nil, errors.New("kubernetes version cannot be nil")
+	}
```

Comment:
We don't perform such validations in other components, so I'd drop this entirely.